### PR TITLE
feat: PBI 13 delete api monitor backend

### PR DESCRIPTION
**Pull request by Exqrch**
### Describe your changes
Create backend to delete monitor api where only the owner can delete them

### Backlog name
PBI-13-delete-api-monitor-backend

### Acceptance criteria
- User can delete API Monitor that is created by them

### Checklist
- [X] I have performed a self-review of my code
- [X] Code successfully run on local
- [X] Feature / changes tested on local
- [X] No failing unit test
- [ ] Passed UAT Test
- [X] Sonarqube tested 
- [ ] Required any changes to environment variable

### CODE REVIEW

File: apimonitor/tests.py [hunk 1]
```diff
@@ -590,3 +590,109 @@ def test_when_authenticated_and_empty_monitor_result_then_return_one_hunderd_suc
...
+
+        # Other user CANNOT delete monitor that is now owned by themself
+        # Get the monitor's ID, in this case: delete first created
+        target_monitor_id = APIMonitor.objects.filter(user=user)[:1].get().id
+        delete_test_path = reverse('monitor-delete-api', kwargs={'pk' : target_monitor_id})
+
+        # Request failed
+        response = self.client.delete(delete_test_path, format='json', **header2)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertEqual(APIMonitor.objects.all().count(), 1)
```
Comment (andrew-susanto) [inline apimonitor/tests.py:692]: gimana kalau test ini (failed) sama test bawahnya (success) dipisah aja luckk, jadi dua buah function terpisah

File: apimonitor/urls.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 router = routers.DefaultRouter()
 router.register(r'', views.APIMonitorViewSet, basename='api-monitor')
-
 urlpatterns = [
     path('', include(router.urls)),
+    path('<int:pk>/delete/', views.monitor_delete_view, name="monitor-delete-api")
```
Comment (andrew-susanto) [inline apimonitor/urls.py:11]: gimana kalau ini digabung sama APIMonitorViewSet aja luckk

File: apimonitor/views.py [hunk 3]
```diff
@@ -16,7 +17,7 @@ class APIMonitorViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
     queryset = APIMonitor.objects.all()
```
Comment (andrew-susanto) [inline apimonitor/views.py:17]: disini bisa tambahin mixins.DestroyModelMix

File: apimonitor/views.py [hunk 4]
```diff
@@ -16,7 +17,7 @@ class APIMonitorViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
...
     serializer_class = APIMonitorSerializer
     permission_classes = [IsAuthenticated]
-    
+
     def get_queryset(self):
         queryset = APIMonitor.objects.filter(user=self.request.user)
         return queryset
```
Comment (andrew-susanto) [inline apimonitor/views.py:23]: lalu def destroy nya jadi disini

File: apimonitor/tests.py (hunk 1 above)
Comment (Exqrch) [reply apimonitor/tests.py]: Sepp, done!

File: apimonitor/views.py (hunk 4 above)
Comment (Exqrch) [reply apimonitor/views.py:24]: Donezo wkwkwk

File: apimonitor/views.py (hunk 3 above)
Comment (Exqrch) [reply apimonitor/views.py:17]: kekw Done

File: apimonitor/urls.py (hunk 2 above)
Comment (Exqrch) [reply apimonitor/urls.py]: Donezo

File: apimonitor/views.py [hunk 5]
```diff
@@ -4,19 +4,21 @@
 from django.db.models import Avg, Count, Q
 from django.utils import timezone
 from django.conf import settings
-from rest_framework import viewsets, mixins
+from rest_framework import viewsets, mixins, generics, status
 from rest_framework.response import Response
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.authtoken.models import Token
```
[the remark this answers is not in the document]
Comment (Exqrch) [reply apimonitor/views.py]: Yep, benar sekali, lupa delete :), thanks

**Review by andrew-susanto — APPROVED** [review]

File: apimonitor/tests.py [hunk 6]
```diff
@@ -590,3 +590,198 @@ def test_when_authenticated_and_empty_monitor_result_then_return_one_hunderd_suc
...
+
+        # Create malicious user
+        malicious_user = User.objects.create_user(username="test2@test.com", email="test2@test.com", password="Test1234")
+        token2 = Token.objects.create(user=malicious_user)
+        header2 = {'HTTP_AUTHORIZATION': f"Token {token2.key}"}
+
+        # Get path
+        target_monitor_id = APIMonitor.objects.filter(user=user)[:1].get().id
+        delete_test_path = reverse('api-monitor-detail', kwargs={'pk' : target_monitor_id})
+        print("DELETE PATH: ", delete_test_path)
```
Comment (Exqrch) [inline apimonitor/tests.py:782]: Print on test

**Review by HugoIr — APPROVED** [review]

**Review by luthfifahlevi — APPROVED** [review]

**Pull request merged**